src/askui/models/openrouter/model.py

```diff
@@ -0,0 +1,159 @@
+import json
+from typing import Optional, Type
+
+from openai import OpenAI
+from typing_extensions import override
+
+from askui.logger import logger
+from askui.models.exceptions import QueryNoResponseError
+from askui.models.models import GetModel
+from askui.models.shared.prompts import SYSTEM_PROMPT_GET
+from askui.models.types.response_schemas import ResponseSchema, to_response_schema
+from askui.utils.image_utils import ImageSource
+
+from .settings import OpenRouterSettings
+
+
+class OpenRouterModel(GetModel):
+    """
+    This class implements the GetModel interface for the OpenRouter API.
+
+    Args:
+        settings (OpenRouterSettings): The settings for the OpenRouter model.
+
+    Example:
+        ```python
+        from askui import VisionAgent
+        from askui.models import (
+            OpenRouterModel,
+            OpenRouterSettings,
+            ModelRegistry,
+        )
+
+
+        # Register OpenRouter model in the registry
+        custom_models: ModelRegistry = {
+            "my-custom-model": OpenRouterGetModel(
+                OpenRouterSettings(
+                    model="anthropic/claude-opus-4",
+                )
+            ),
+        }
+
+        with VisionAgent(models=custom_models, model={"get":"my-custom-model"}) as agent:
+            result = agent.get("What is the main heading on the screen?")
+            print(result)
+        ```
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        settings: OpenRouterSettings | None = None,
+        client: Optional[OpenAI] = None,
+    ):
+        self._settings = settings or OpenRouterSettings()
+
+        self._client = (
+            client
+            if client is not None
+            else OpenAI(
+                api_key=self._settings.open_router_api_key.get_secret_value(),
+                base_url=str(self._settings.base_url),
+            )
+        )
+
+    def _predict(
+        self,
+        image_url: str,
+        instruction: str,
+        prompt: str,
+        response_schema: type[ResponseSchema] | None,
+    ) -> str | None | ResponseSchema:
+        extra_body: dict[str, object] = {}
+
+        if len(self._settings.models) > 0:
+            extra_body["models"] = self._settings.models
+
+        response_format: dict[str, object] | None = None
+        if response_schema is not None:
+            extra_body["provider"] = {"require_parameters": True}
+
+            _response_schema = to_response_schema(response_schema)
+            response_format = {
+                "type": "json_schema",
+                "json_schema": {
+                    "name": "user_json_schema",
+                    "schema": next(
+                        iter(
+                            _response_schema.model_json_schema()
+                            .get("$defs", {})
+                            .values()
+                        )
+                    ),
```

Comment:
What do we do here? Does that also include the schemas not defined under `"$defs"` as `"$defs"` is only used for nested models as far as I know.